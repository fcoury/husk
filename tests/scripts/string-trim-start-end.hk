// Test trim_start() and trim_end() methods for strings
let s1 = "  hello  ";
let s2 = "\t\nworld\r\n";
let s3 = "   ";
let s4 = "no whitespace";
let s5 = "  multiple   spaces  in   middle  ";

// Test trim_start
println!(s1.trim_start());               // Should print: "hello  "
println!(s2.trim_start());               // Should print: "world\r\n"
println!(s3.trim_start());               // Should print: ""
println!(s4.trim_start());               // Should print: "no whitespace"
println!(s5.trim_start());               // Should print: "multiple   spaces  in   middle  "

// Test trim_end
println!(s1.trim_end());                 // Should print: "  hello"
println!(s2.trim_end());                 // Should print: "\t\nworld"
println!(s3.trim_end());                 // Should print: ""
println!(s4.trim_end());                 // Should print: "no whitespace"
println!(s5.trim_end());                 // Should print: "  multiple   spaces  in   middle"

// Test chaining
let s6 = "   both sides   ";
println!(s6.trim_start().trim_end());    // Should print: "both sides"
println!(s6.trim_end().trim_start());    // Should print: "both sides"

// Test with Unicode whitespace
let s7 = "\u{00A0}test\u{00A0}";       // Non-breaking space
println!(s7.trim_start());               // Should handle Unicode whitespace
println!(s7.trim_end());                 // Should handle Unicode whitespace

// Test length after trimming
let s8 = "  test  ";
println!(s8.len());                      // Should print: 8
println!(s8.trim_start().len());         // Should print: 6
println!(s8.trim_end().len());           // Should print: 6
println!(s8.trim_start().trim_end().len()); // Should print: 4

// Test on empty string
let empty = "";
println!(empty.trim_start());            // Should print: ""
println!(empty.trim_end());              // Should print: ""
println!(empty.trim_start().is_empty()); // Should print: true