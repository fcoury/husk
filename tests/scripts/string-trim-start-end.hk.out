hello  
world


no whitespace
multiple   spaces  in   middle  
  hello
	
world

no whitespace
  multiple   spaces  in   middle
both sides
both sides
\u{00A0}test\u{00A0}
\u{00A0}test\u{00A0}
8
6
6
4


true
